services/wallet/balance_cache.go

```diff
@@ -0,0 +1,55 @@
+package wallet
+
+import (
+	"context"
+	"math/big"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type BalanceCache struct {
```

Comment:
Looks like we don't need to export it:
```suggestion
type balanceCache struct {
```